src/types/messages.ts

```diff
@@ -94,11 +93,14 @@ export type TUserMessage<Metadata = TMessageMetadata> = TBaseMessage<Metadata> &
     avatarUrl?: string;
 };
 
-export type TAssistantMessage<Metadata = TMessageMetadata> = TBaseMessage<Metadata> & {
+export type TAssistantMessage<
+    Metadata = TMessageMetadata,
+    TAdditionalPart extends TMessagePart = never,
+> = TBaseMessage<Metadata> & {
     role: 'assistant';
-    content: string | TMessagePart | TMessagePart[];
+    content: TAssistantMessageContent<TAdditionalPart>;
 };
 
-export type TMessage<Metadata = TMessageMetadata> =
+export type TMessage<TAdditionalPart extends TMessagePart = never, Metadata = TMessageMetadata> =
```

Comment:
TAdditional look very common and not understand why it's need
maybe
```suggestion
export type TMessage<TCustomMessageContent extends TMessagePart = never, Metadata = TMessageMetadata> =
```

Reply:
yep, renamed types